test/test_profiler.py

```diff
@@ -80,7 +80,17 @@ def train_worker():
         duration_ms=5000,
         num_tracing_attempts=5,
         delay_ms=1000)
+
+    with self.assertRaisesRegex(RuntimeError, "unimplemented"):
+      xp.monitor(f'localhost:{port}', duration_ms=2000, monitoring_level=2)
+
     p.terminate()
+
+    with self.assertRaisesRegex(RuntimeError,
```

Comment:
Just ran your tests locally. They are great for checking the monitor API call correctness.

Please add a test to cover a successful run of `xp.monitor()`.

Reply:
Hi @kat25230. Checking in. How did the test go for you?

Reply:
@miladm Thanks for checking in, I didn't have time to work on it last week. Hopefully will be able to get to it this week. :)

Reply:
@kat25230 
Thanks for the update! Looking forward to your update. :)

Reply:
Hi @kat25230 checking in. How is the test going for you?

Reply:
Hi @miladm sorry for the delay, was really busy with and didn't find the time to finish this. 
Will have a test for you by the end of the week :)